frontend/src/App.js

```diff
@@ -3,41 +3,56 @@ import axios from 'axios';
 import logo from './logo.svg';
 import './App.css';
 
+const dataBlockDuration = 5; // seconds
+
 function App() {
   const [tubeData, setTubeData] = useState([]);
 
   const lines = "bakerloo,central,circle,district,hammersmith-city,jubilee,metropolitan,northern,piccadilly,victoria,waterloo-city";
 
   useEffect(() => {
-    axios.get(`https://api.tfl.gov.uk/Line/${lines}/Arrivals?`)
-      .then(response => {
-        const data = response.data;
-        const filteredData = data
-          .filter(item => item.timeToStation < 60)
-          .map(item => ({
-            id: item.id,
-            stationName: item.stationName,
-            lineName: item.lineName,
-            timeToStation: item.timeToStation
-          }));
-        const sortedData = filteredData.sort((a, b) => a.timeToStation - b.timeToStation);
-        setTubeData(sortedData);
-        // Display tube data
-        console.log(sortedData);
-        // Save sortedData to localStorage
-        // localStorage.setItem('sortedData', JSON.stringify(sortedData));
-      })
-      .catch(error => {
-        console.error('Error fetching tube data:', error);
-      });
+    const fetchData = () => {
+      axios.get(`https://api.tfl.gov.uk/Line/${lines}/Arrivals?`)
+        .then(response => {
+          const data = response.data;
+          const filteredData = data
+            .filter(item => item.timeToStation < dataBlockDuration)
+            .map(item => ({
+              id: item.id,
+              stationName: item.stationName,
+              lineName: item.lineName,
+              timeToStation: item.timeToStation
+            }));
+          const sortedData = filteredData.sort((a, b) => a.timeToStation - b.timeToStation);
+          setTubeData(sortedData);
+          // Save sortedData to localStorage
+          // localStorage.setItem('sortedData', JSON.stringify(sortedData));
+        })
+        .catch(error => {
+          console.error('Error fetching tube data:', error);
+        });
+    };
+
+    // Fetch data on first render
+    fetchData();
+
+    // Fetch data every <dataBlockDuration> seconds
+    const interval = setInterval(fetchData, (dataBlockDuration * 1000));
+
+    // Clean up the interval on component unmount
+    return () => clearInterval(interval);
```

Comment:
interested in where this function comes from? Its not defined anywhere or imported